mods/fire/init.lua

```diff
@@ -78,13 +78,19 @@ minetest.register_tool("fire:flint_and_steel", {
 		local pt = pointed_thing
 
 		if pt.type == "node" and minetest.get_node(pt.above).name == "air" then
+			local node_under = minetest.get_node(pt.under).name
+			if minetest.get_node_group(node_under, "flammable") == 0 or
+					minetest.registered_nodes[node_under].drawtype ~= "normal" then
+				return
```

Comment:
I think all attempts at ignition (on flammable or non flammable nodes) should add wear to the flint and steel, so this check can be included below, no 'return'.
We could always increase the number of uses to compensate.